### ISSUE TITLE
Interesting papers (models & losses) 

### ISSUE THREAD

**Issue by rpytel1** [rpytel1/efficient-semantic-segmentation#4]
# 
Interesting resources with an intro to image segmentation field:
- [Image segmentation in 2020: Architectures, Losses, Datasets, and Frameworks](https://neptune.ai/blog/image-segmentation-in-2020) 
- [A 2019 Guide to Semantic Segmentation](https://heartbeat.fritz.ai/a-2019-guide-to-semantic-segmentation-ca8242f5a7fc)

### UNet (MICCAI, 2015)
- already implemented in fast.ai
- one of the most popular models in Image Segmentation
- kind of old architecture

### Tiramisu (2017)
- code available in fast.ai course materials
- 88 % accuracy on Camvid

### FastFCN(2019)
- newer architecture
- code available (Pytorch code: https://github.com/wuhuikai/FastFCN)
- fast.ai code: TBA
- fully connected network with joint pyramid upsampling

### Gated-SCNN(2019)
- takes two types of input (casual image and shapes)
- one of the newest architectures
- code (Pytorch implementation: https://github.com/nv-tlabs/GSCNN)
- fast..ai code: TBA

### Focal loss
- improvement of cross-entropy loss
- Pytorch implementation (https://www.kaggle.com/c/tgs-salt-identification-challenge/discussion/65938)

### Dice loss
- most commonly used loss in Image Segmentation
- Pytorch code (https://github.com/kevinzakka/pytorch-goodies/blob/master/losses.py)

### Boundary loss
- advised for unbalanced datasets 
- **Pytorch implementation of this and other losses**  (https://github.com/JunMa11/SegLoss/tree/master/losses_pytorch)

### Lovász-Softmax loss
- This loss performs direct optimization of the mean intersection-over-union loss in neural networks based on the convex Lovasz extension of sub-modular losses.
- Pytorch implementation (https://github.com/bermanmaxim/LovaszSoftmax/blob/master/pytorch/lovasz_losses.py)
